src/redux/api/api.js

```diff
@@ -0,0 +1,33 @@
+import library from '../../api/library';
+
+const RECEIVE_BOOKS = 'bookStore/books/RECEIVE_BOOKS';
+
+export const receiveBooks = (books) => ({
+  type: RECEIVE_BOOKS,
+  books,
+});
+
+export const getBooks = () => (dispatch) => {
+  library.getBooks().then((books) => {
+    dispatch(receiveBooks(books));
+    return books;
+  });
+};
+
+export const addBook = (book) => (dispatch) => {
+  library.addBook(book).then((response) => {
+    if (response === 'Created') {
+      dispatch(getBooks());
+    }
+    return response;
+  });
+};
+
+export const removeBook = (book) => (dispatch) => {
+  library.removeBook(book).then((response) => {
+    if (response === 'The book was deleted successfully!') {
+      dispatch(getBooks());
+    }
+    return response;
+  });
```

Comment:
- [ ] OPTIONAL - Consider fixing an issue that would occur here, after filtering a book and clicking the remove button on the filtered books the book still remains on the dom. It only occurs after filtering a book. Consider updating the filtered books state if it isn't in sync with the actual books state.

Reply:
:ok_man:  Ok, Thanks for pointing it out, I didn't realize about it before. I have already add a new case in filter reducer so it can process the updated book arrays.